compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -1645,7 +1645,9 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
 
         let tcx = self.infcx.tcx;
         let (def_id, substs) = match *placeholder_trait_predicate.trait_ref.self_ty().kind() {
-            ty::Alias(_, ty::AliasTy { def_id, substs, .. }) => (def_id, substs),
```

Comment:
This line caused #111399. Not sure if the other `match`es would've exhibited ICEs at some point or not but in any case this PR makes them more correct™.